### ISSUE TITLE
Rest parameter should not trigger javascript:S1788

### ISSUE THREAD

**Issue by hcomnetworkers** [SonarSource/SonarJS#1120]
**I want to report a bug.**

**SonarJS version:** 5.0<br>
**SonarQube version:** Community EditionVersion 7.2.1 (build 14109)<br>

**Rule key:** javascript:S1788

**Reproducer** 
```
export function myFn(x = 1, ...rest) {}
```

The rest parameter cannot have a default value and the `x` cannot be moved last.

**Comment by vilchik-elena** [comment]
https://github.com/SonarSource/eslint-plugin-sonarjs/issues/89 should be implemented keeping in mind this FP. When https://github.com/SonarSource/eslint-plugin-sonarjs/issues/89 will replace current implementation, FP will be fixed.
